src/main/java/com/nitnelave/CreeperHeal/block/CreeperExplosion.java

```diff
@@ -257,57 +256,77 @@ private boolean isObsidianLike(Material m, boolean table)
      */
     public void record(Block block)
     {
-    	if (block.getType() == Material.PORTAL)
-    		return;
-    	
+        if (block.getType() == Material.PORTAL || checked.contains(new ShortLocation(block)))
+            return;
+
         CreeperBlock cBlock = CreeperBlock.newBlock(block.getState());
 
-        if (cBlock == null || checked.contains(new ShortLocation(block)))
+        if (cBlock == null)
             return;
 
-        checked.add(new ShortLocation(block));
+        record(cBlock);
+    }
 
-        if ((CreeperConfig.getBool(CfgVal.PREVENT_CHAIN_REACTION) && block.getType().equals(Material.TNT))
-            || world.isProtected(block))
+    /**
+     * Add a Replaceable to the list, and remove it from the world.
+     *
+     * @param replaceable
+     *            The Replaceable to add.
+     */
+    public void record(Replaceable replaceable)
+    {
+        if (replaceable == null || replaceable.getType() == Material.PORTAL)
+            return;
+
+        ShortLocation location = new ShortLocation(replaceable.getLocation());
+
+        if (checked.contains(location))
```

Comment:
I wouldn't add non-block entities to the checked location, since they can easily overlap with something else, or each other.